browser/src/map/Clipboard.js

```diff
@@ -956,32 +964,75 @@ L.Clipboard = L.Class.extend({
 		};
 	},
 
-	// Executes the navigator.clipboard.read() call, if it's available.
-	_navigatorClipboardRead: function(isSpecial) {
-		if (!L.Browser.clipboardApiAvailable) {
-			return false;
-		}
+	_asyncAttemptNavigatorClipboardRead: async function(isSpecial) {
+		let clipboard = navigator.clipboard;
 
-		var that = this;
-		var clipboard = navigator.clipboard;
 		if (L.Browser.cypressTest) {
 			clipboard = this._dummyClipboard;
 		}
-		clipboard.read().then(function(clipboardContents) {
-			if (isSpecial) {
-				that._navigatorClipboardPasteSpecial = true;
-			}
-			that._navigatorClipboardReadCallback(clipboardContents);
-		}, function(error) {
-			window.app.console.log('navigator.clipboard.read() failed: ' + error.message);
+
+		let clipboardContents;
+
+		try {
+			clipboardContents = await clipboard.read();
+		} catch (error) {
+			window.app.console.log(
+				'navigator.clipboard.read() failed: ' + error.message,
+			);
 			if (isSpecial) {
 				// Fallback to the old code, as in filterExecCopyPaste().
-				that._openPasteSpecialPopup();
+				this._openPasteSpecialPopup();
 			} else {
 				// Fallback to the old code, as in _execCopyCutPaste().
-				that._afterCopyCutPaste('paste');
+				this._afterCopyCutPaste('paste');
 			}
-		});
+			return;
+		}
+
+		if (isSpecial) {
+			this._navigatorClipboardPasteSpecial = true;
+		}
+
+		if (clipboardContents.length < 1) {
```

Comment:
I think it was better as it was (in separate method: `_navigatorClipboardReadCallback`) - > shorter function to read.